modules/shared/programs/claude/files/skills/analyzing-da-sessions/scripts/analyze.py

```
@@ -810,6 +849,62 @@ def fetch_remote_file(host: str, path: str, warnings: list[str]) -> str | None:
 
+def check_controlmaster_active(host: str, warnings: list[str]) -> bool:
+    """ControlMaster master socket이 활성인지 확인. 비활성이면 master 생성을 1회 시도 후 재확인.
+
+    `ssh -O check <host>`는 master 부재 시 실패한다. 따라서 실패 시 일반 `ssh <host> true`로
+    master 생성 시도 후 다시 `-O check`로 확인하는 2단계 sequence로 구성한다.
+
+    返回值이 False이면 worker pool은 K=1로 강등된다 (degrade fallback).
+    """
+    _validate_host(host)
+    try:
+        proc = subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)